markdown/usage/nf_core_tutorial.md

```diff
@@ -288,63 +397,132 @@ nf-core lint /path/to/pipeline
+Each file Each file (`branch.yml`, `ci.yml` and `linting.yml`) defines several tests: branch protection, running the pipeline with the test data, linting the code with `nf-core lint`, linting the syntax of all Markdown documentation and linting the syntax of all YAML files.
 
-When writing `conf/test.config` remember to define all required parameters so that the pipeline will run with only `-profile test`. Note that remote URLs cannot be traversed like a regular file system - so glob file expansions such as `*.fa` will not work.
+#### `branch.yml`
 
-### Travis CI configuration
-The automated tests with Travis CI are configured in the `.travis.yml` file that is generated by the template. The `script` block defines three tests: linting the code with `nf-core lint`, linting the syntax of all Markdown documentation and running the pipeline with the test data.
+This is already set up and should not be touched.
+It will check that a PR is made on `dev` unless it's for a release or a patch.
 
-The `env` section sets the `NXF_VER` environment variable twice. This tells Travis to run the tests twice in parallel - once with the latest version of Nextflow (`NXF_VER=''`) and once with the minimum version supported by the pipeline. Do not edit this version number manually - it appears in multiple locations through the pipeline code, so it's better to use `nf-core bump-version --nextflow` instead.
+#### `linting.yml`
 
-The provided tests may be sufficient for your pipeline. However, if it is possible to run the pipeline with significantly different options (for example, different alignment tools), then it is good to test all of these. You can do this by adding additional commands in the `script` block.
+This is already set up and should not be touched.
+It will check:
+
+* That all Markdown documentation follow a proper syntax.
+(if you're using `atom` of `Visual Studio Code` or any similar editor, plugins are available to help you out: _e.g._ [markdownlint for atom](https://atom.io/packages/linter-markdownlint), [markdownlint for Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint))
+* That all YAML file follow a proper syntax.
+* That the code follow the nf-core linting.
+
+#### `ci.yml`
+
+The `nxf_ver` variable sets the `NXF_VER` environment variable twice.
+This tells GitHub Actions to run the tests twice in parallel - once with the latest version of Nextflow (`NXF_VER=''`) and once with the minimum version supported by the pipeline.
+Do not edit this version number manually - it appears in multiple locations through the pipeline code, so it's better to use `nf-core bump-version --nextflow` instead.
+
+The provided tests may be sufficient for your pipeline.
+However, if it is possible to run the pipeline with significantly different options (for example, different alignment tools), then it is good to test all of these.
+You can do this by adding additional tests in the `jobs` block.
```

Comment:
```suggestion
You can do this by adding additional tests in the `jobs` block.
Do not try add a run for every possible combination of parameters, as it will take too long to run.
```